subsys/net/lib/mqtt_socket/mqtt.c

```diff
@@ -0,0 +1,699 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt.c
+ *
+ * @brief MQTT Client API Implementation.
+ */
+
+#define LOG_MODULE_NAME mqtt
```

Comment:
I suggest that we prefix this with "net_" like in other network related libraries, that way it is possible to set runtime logging filters by just "net*" that would catch all the network related output. Please do that for all the other .c files in this PR too.